tools/mtouch/mtouch.cs

```diff
@@ -1331,16 +1336,27 @@ static int Main2 (string[] args)
 					app.UseInterpreter = false;
 				}
 
-				// FIXME: the interpreter only supports ARM64 right now
+				// FIXME: the interpreter only supports ARM64{,_32} right now
 				// temporary - without a check here the error happens when deploying
-				if (!app.IsSimulatorBuild && !app.IsArchEnabled (Abi.ARM64))
+				if (!app.IsSimulatorBuild && (!app.IsArchEnabled (Abi.ARM64) && !app.IsArchEnabled (Abi.ARM64_32)))
 					throw ErrorHelper.CreateError (99, "Internal error: The interpreter is currently only available for 64 bits.");
 
 				// needs to be set after the argument validations
 				// interpreter can use some extra code (e.g. SRE) that is not shipped in the default (AOT) profile
 				app.EnableRepl = true;
+			} else {
+				if (app.Platform == ApplePlatform.WatchOS && app.IsArchEnabled (Abi.ARM64_32)) {
+					if (app.IsArchEnabled (Abi.ARMv7k)) {
+						throw ErrorHelper.CreateError (99, "Please use device builds on WatchOS.");
```

Comment:
I didn't manage to disable device-specific builds, but I believe this would be the condition.

Also: To what does the error code maps to? Is there some guideline?

Reply:
mtouch error codes are listed and documented here: https://github.com/xamarin/xamarin-macios/blob/master/docs/website/mtouch-errors.md#mt0142-cannot-find-the-assembly-assembly-passed-as-an-argument-to---interpreter

it looks like the next available code would be `143`.

Also you're not checking for debug mode here, so this will trigger for release builds as well.